### PR TITLE
Get sort order working correctly again. 

### PR DESCRIPTION
**Pull request by Camwyn**
Add a `menu_order` attribute to the ticket object, and functions for dealing with it.

Remove an unused js function.

Fix a few js errors - bad dependencies in the enqueue.
Also a bit of cleanup a-la phpcs (rename unused variables, whitespace)

Ticket: https://central.tri.be/issues/80617

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:413](https://github.com/moderntribe/event-tickets/blob/c479b91a24086990766419141e3bb472a8b5309e/src/Tribe/Tickets_Handler.php#L413)
:x: | Visibility must be declared on method &quot;sortTicketsByMenuOrder&quot; | [Tickets_Handler.php:1105](https://github.com/moderntribe/event-tickets/blob/c479b91a24086990766419141e3bb472a8b5309e/src/Tribe/Tickets_Handler.php#L1105)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:413](https://github.com/moderntribe/event-tickets/blob/afb38b1140226b8f68a805c19287c2ad04b45934/src/Tribe/Tickets_Handler.php#L413)
:x: | Visibility must be declared on method &quot;sortTicketsByMenuOrder&quot; | [Tickets_Handler.php:1104](https://github.com/moderntribe/event-tickets/blob/afb38b1140226b8f68a805c19287c2ad04b45934/src/Tribe/Tickets_Handler.php#L1104)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:413](https://github.com/moderntribe/event-tickets/blob/a68ffd481a262fe73f7b11f2d4e060630e7f4907/src/Tribe/Tickets_Handler.php#L413)
:x: | Visibility must be declared on method &quot;sortTicketsByMenuOrder&quot; | [Tickets_Handler.php:1104](https://github.com/moderntribe/event-tickets/blob/a68ffd481a262fe73f7b11f2d4e060630e7f4907/src/Tribe/Tickets_Handler.php#L1104)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:423](https://github.com/moderntribe/event-tickets/blob/162bdff5cfcaef794630071e55979f11021f7d77/src/Tribe/Tickets_Handler.php#L423)
:x: | Visibility must be declared on method &quot;sortTicketsByMenuOrder&quot; | [Tickets_Handler.php:1115](https://github.com/moderntribe/event-tickets/blob/162bdff5cfcaef794630071e55979f11021f7d77/src/Tribe/Tickets_Handler.php#L1115)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets_Handler.php [hunk 1]
```diff
@@ -93,9 +98,10 @@ public function __construct() {
 		foreach ( $main->post_types() as $post_type ) {
 			add_action( 'save_post_' . $post_type, array( $this, 'save_image_header' ) );
 			add_action( 'save_post_' . $post_type, array( $this, 'save_global_stock' ) );
-			add_action( 'save_post_' . $post_type, array( $this, 'save_tickets_order' ) );
 		}
 
+		add_action( 'save_post_tribe_events', array( $this, 'save_tickets_order' ) );
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: prior to this change the `save_tickets_order` method was hooked to `save_post_{post_type}` [line `96`]

 is it **intentional** to unhook this method from all `save_post_{post_type}` hooks except `tribe_events`?

File: src/Tribe/Tickets_Handler.php [hunk 2]
```diff
@@ -559,7 +566,7 @@ public function attendees_admin_body_class( $body_classes ) {
 	 *
 	 * @return string
 	 */
-	public function attendees_admin_title( $admin_title, $title ) {
+	public function attendees_admin_title( $admin_title, $unused_title ) {
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: would encourage changing the `@param $title` section to be `@param $unused_title` to match the change in param name here

File: src/Tribe/Tickets_Handler.php [hunk 3]
```diff
@@ -995,7 +1002,7 @@ public function get_header_image_id( $event_id ) {
 	 * @param int $post_id
 	 */
 	public function save_image_header( $post_id ) {
-		if ( ! ( isset($_POST[ 'tribe-tickets-post-settings' ])  && wp_verify_nonce( $_POST[ 'tribe-tickets-post-settings' ], 'tribe-tickets-meta-box' ) ) ) {
+		if ( ! ( isset( $_POST[ 'tribe-tickets-post-settings' ] )  && wp_verify_nonce( $_POST[ 'tribe-tickets-post-settings' ], 'tribe-tickets-meta-box' ) ) ) {
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: two spaces between `)` and `&&` here — change to one space

File: src/Tribe/Tickets_Handler.php [hunk 4]
```diff
@@ -1053,23 +1062,37 @@ public function save_tickets_order( $post_id ) {
...
+			$ticket_order
+		);
+
+		$ticket_order = explode( ',', $ticket_order );
+		$ticket_order = array_flip( $ticket_order );
+
+		foreach ( $ticket_order as $id => $order ) {
+			wp_update_post( array(
+				'ID'           => absint( $id ),
+				'menu_order'   => absint( $order ),
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: there are two spaces between `'menu_order'` and  `=>` here — reduce to one space, and align the spacing of `'ID'` and _its_ `=>` accordingly

File: src/Tribe/Tickets_Handler.php [hunk 5]
```diff
@@ -1053,23 +1062,37 @@ public function save_tickets_order( $post_id ) {
...
+
 		return;
 	}
+
 	/**
 	 * Adds the hidden input to store the drag-n-drop ticket order
+	 *
+	 * @param int $post_id
+	 *
+	 * @since TBD
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: the `@since` should be above `@param`

File: src/Tribe/Tickets_Handler.php [hunk 6]
```diff
@@ -1079,6 +1102,32 @@ public function tickets_order_input( $post_id ) {
...
+	 *
+	 * @param array $tickets array of ticket objects
+	 *
+	 * @return array - sorted array of ticket objects
+	 */
+	public function sort_tickets_by_menu_order( $tickets ) {
+		// our sort function
+		function sortTicketsByMenuOrder( $a, $b ) {
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: alter this name to be snake_cased — `_sort_tickets_by_menu_order_callback` or something

File: src/Tribe/Tickets_Handler.php [hunk 7]
```diff
@@ -1079,6 +1102,32 @@ public function tickets_order_input( $post_id ) {
...
+			return $a->menu_order - $b->menu_order;
+		}
+
+		foreach ( $tickets as $key => $ticket ) {
+			// make sure they are ordered correctly
+			$orderpost = get_post( $ticket->ID );
+			$ticket->menu_order = $orderpost->menu_order;
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: vertically align the `=` for these assignments

File: src/Tribe/Tickets_Handler.php (hunk 1 above)
Comment (Camwyn) [inline src/Tribe/Tickets_Handler.php]: Yes - it's not supposed to trigger on ticket save, only on post save.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:423](https://github.com/moderntribe/event-tickets/blob/7281043c3776a6bb1a8e018055a8110de104500d/src/Tribe/Tickets_Handler.php#L423)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:423](https://github.com/moderntribe/event-tickets/blob/ae31c92e30f733b6f7c14824d1dd566eb4f10fac/src/Tribe/Tickets_Handler.php#L423)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:423](https://github.com/moderntribe/event-tickets/blob/e2d9ca3e4358db415ce0534e2438aa829e99a304/src/Tribe/Tickets_Handler.php#L423)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:423](https://github.com/moderntribe/event-tickets/blob/6bc13f0bb075a1bf119d00b8b2b29d907342e833/src/Tribe/Tickets_Handler.php#L423)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:423](https://github.com/moderntribe/event-tickets/blob/23bf7da17e6814b028583de97a20f14c8b06b77e/src/Tribe/Tickets_Handler.php#L423)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets_Handler.php [hunk 8]
```diff
@@ -93,9 +98,10 @@ public function __construct() {
 		foreach ( $main->post_types() as $post_type ) {
 			add_action( 'save_post_' . $post_type, array( $this, 'save_image_header' ) );
 			add_action( 'save_post_' . $post_type, array( $this, 'save_global_stock' ) );
-			add_action( 'save_post_' . $post_type, array( $this, 'save_tickets_order' ) );
 		}
 
+		add_action( 'save_post_' . Tribe__Events__Main::POSTTYPE, array( $this, 'save_tickets_order' ) );
```
Comment (ggwicz) [inline src/Tribe/Tickets_Handler.php]: is removing this method from all tickets-enabled post types other than `tribe_events` intentional? 

i reached out to @Camwyn on slack for further clarification on this ...

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_Handler.php:413](https://github.com/moderntribe/event-tickets/blob/5e68946173326a425dcd5c2f382d466bf44b1b35/src/Tribe/Tickets_Handler.php#L413)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets_Handler.php (hunk 8 above)
Comment (Camwyn) [inline src/Tribe/Tickets_Handler.php]: That was not the intention, was done in response to some looping issues. Corrected.

**Pull request merged**